### ISSUE TITLE
WindowsServerStandard/hyper-v: Timeout during vagrant up

### ISSUE THREAD

**Issue by gerardog** [cdaf/windows#10]
First, thank you very much for providing this image. First one I tried that has "Install-WindowsFeature" working!
Now the question: Should the vagrant provisioning step map a network drive and return some OK status?
I am getting a timeout instead.
Using vagrant  2.2.1 + Windows 10 [Version 10.0.15063] 

```
$ vagrant init cdaf/WindowsServerStandard
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.

$ vagrant up
Bringing machine 'default' up with 'hyperv' provider...
==> default: Verifying Hyper-V is enabled...
==> default: Verifying Hyper-V is accessible...
==> default: Importing a Hyper-V instance
    default: Creating and registering the VM...
    default: Successfully imported VM
    default: Please choose a switch to attach to your Hyper-V instance.
    default: If none of these are appropriate, please open the Hyper-V manager
    default: to create a new virtual switch.
    default:
    default: 1) Internal
    default:
    default: What switch would you like to use? 1
    default: Configuring the VM...
==> default: Starting the machine...
==> default: Waiting for the machine to report its IP address...
    default: Timeout: 120 seconds
    default: IP: 192.168.137.118
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 192.168.137.118:22
    default: SSH username: vagrant
    default: SSH auth method: private key
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.

$ vagrant rdp
VM must be created before running this command. Run `vagrant up` first.

$ vagrant ssh

$ (no response whatsoever)

```

I was able to connect using Hyper-v console. Was forced to use administrator user and guessed that the password was 'vagrant'. Installing windows features worked. To connect remotely I had to disable the Windows Firewall using:

Set-NetFirewallProfile -Profile Domain,Public,Private -Enabled False

After turning off the firewall, I tried 'vagrant halt' and 'vagrant up' but I got the same error.
What is the expected result?

**Comment by gerardog** [comment]
Looks like the image does not have an SSH Server installed. 
That may be ok for many, but it would be nice to know that upfront. Users should not expect 'vagrant rdp', 'vagrant ssh' or shared folders to work out of the box.

I tried to setup the SSH server myself, but no luck yet. 
```
choco install openssh -params '"/SSHServerFeature /KeyBasedAuthenticationFeature"' -y
md C:\Users\vagrant/.ssh 

$client = new-object System.Net.WebClient
$client.DownloadFile("https://raw.githubusercontent.com/hashicorp/vagrant/master/keys/vagrant.pub","C:\Users\vagrant\.ssh\id_rsa.pub")
$client.DownloadFile("https://raw.githubusercontent.com/hashicorp/vagrant/master/keys/vagrant","C:\Users\vagrant\.ssh\id_rsa")
```
But the ssh connection failed anyway (authentication failure). So switched to password-auth editing 'vagrantfile'
```
  config.ssh.username="vagrant"
  config.ssh.password="vagrant"
```
And now I got a different error:

```
$ vagrant up
(...)
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 10.10.6.79:22
    default: SSH username: vagrant
    default: SSH auth method: password
The configured shell (config.ssh.shell) is invalid and unable
to properly execute commands. The most common cause for this is
using a shell that is unavailable on the system. Please verify
you're using the full path to the shell and that the shell is
executable by the SSH user.
```
I can SSH into the box, and a regular CMD.EXE shell opens. not sure what's wrong there.

**Comment by gerardog** [comment]
Tried:
config.vm.communicator="winrm"
Now I got:

```
$ vagrant up
Bringing machine 'default' up with 'hyperv' provider...
==> default: Verifying Hyper-V is enabled...
==> default: Verifying Hyper-V is accessible...
    default: Configuring the VM...
==> default: Starting the machine...
==> default: Waiting for the machine to report its IP address...
    default: Timeout: 120 seconds
    default: IP: 10.10.6.79
==> default: Waiting for machine to boot. This may take a few minutes...
    default: WinRM address: 10.10.6.79:5985
    default: WinRM username: vagrant
    default: WinRM execution_time_limit: PT2H
    default: WinRM transport: negotiate
==> default: Machine booted and ready!
==> default: Preparing SMB shared folders...
    default: You will be asked for the username and password to use for the SMB
    default: folders shortly. Please use the proper username/password of your
    default: account.
    default:
    default: Username: JohnDoe
    default: Password (will be hidden):

Vagrant requires administrator access to create SMB shares and
may request access to complete setup of configured shares.

Exporting an SMB share failed! Details about the failure are shown
below. Please inspect the error message and correct any problems.

Host path:

Stderr:

Stdout:
```

**Comment by gerardog** [comment]
Which is a vagrant 2.2.0 issue: see https://github.com/hashicorp/vagrant/issues/10438

**Comment by gerardog** [comment]
Rolled back to vagrant 2.1.5 and it worked!

First I had another vagrant issue which I worked around: https://github.com/hashicorp/vagrant/issues/10175#issuecomment-442245139

Then used `config.vm.communicator="winrm"` and Vagrant mounted the SMB shared folder, and 'vagrant rdp' also worked. 'vagrant ssh' did not.

**Comment by cdaf** [comment]
My bad, I only set the required WinRM configs in the VirtualBox instance, I will work on getting to use WinRM by default in the Hyper-V image this evening.

**Comment by cdaf** [comment]
In the interim, if you add these to your Vagrantfile it should work until I get it set by default https://github.com/cdaf/windows/blob/master/samples/vagrant-box/Vagrantfile

**Comment by cdaf** [comment]
apologies for the slow response, this has proven to be more difficult to address that I had thought, will keep you in the loop when I make progress.

**Comment by cdaf** [comment]
ah, I've found the setting and updated Hyper-V specific [ip_address_timeout](https://stackoverflow.com/questions/43175806/vagrant-windows-on-windows-config-vm-boot-timeout-not-applied), this will take a little while to regression test, upload and publish.

**Comment by cdaf** [comment]
I've released version 1.8.8 of this box to address this issue. It also includes the January cumulative update.
https://app.vagrantup.com/cdaf/boxes/WindowsServerStandard
